middleware.js

```diff
@@ -0,0 +1,211 @@
+/**                   _
+ *  _             _ _| |_
+ * | |           | |_   _|
+ * | |___  _   _ | | |_|
+ * | '_  \| | | || | | |
+ * | | | || |_| || | | |
+ * |_| |_|\___,_||_| |_|
+ *
+ * (c) Huli Inc
+ */
+
+
+/**
+ * @file Manages and executes middlewares in all the different app phases
+ * @requires ju-shared/observable-class
+ * @module ju-mvc/middleware
+ * @extends ju-shared/observable-class
+ */
+
+define([
+	'ju-shared/observable-class'
+],
+function (
+	ObservableClass
+){
+	'use strict';
+
+
+	/** Constants **/
+
+    /**
+     * @constant {String} DEFAULT_SUBPHASE - if the subphase is not provided when adding a middleware this will be use
+     */
+	var DEFAULT_SUBPHASE = 'during';
+
+    /**
+     * Object used to store all the phases and middlewares
+     * @type {Object}
+     */
+    var phases = {};
```

Comment:
If Middleware is a singleton, then why storing phases outside the instance? this.phases?